blaze/compute/sql.py

```diff
@@ -585,6 +592,17 @@ def compute_up(expr, data, **kwargs):
     return table_of_engine(data, expr._name)
 
 
+@dispatch(DateTime, (ClauseElement, sa.sql.elements.ColumnElement))
+def compute_up(expr, data, **kwargs):
+    return getattr(sa.func, expr.attr)(data).label(expr._name)
```

Comment:
Actually, I hadn't seen that! Excellent. Thanks. So I should be able to do something like this:

1. blaze will compile everything to an `extract` call
2. then we compile extract to different things based on the dialect (default does whatever the default of sqlalchemy does, version below compiles to a function call)

```python
@compiles(Extract, 'mysql', 'hive', 'mssql')
def compile_date_extract(element, compiler, **kwargs):
    return compiler.visit_function(getattr(sa.func, element.field)(ex.expr.effective_value), **kwargs)
```